event_profiling.c

```diff
@@ -0,0 +1,415 @@
+#include "event_profiling.h"
+
+/* Ruby */
+#include "ractor_core.h"
+#include "ruby.h"
+#include "ruby/atomic.h"
+#include "ruby/thread_native.h"
+#include "vm_core.h"
+
+#if USE_EVENT_PROFILING
+
+/* GCC warning */
+#include <sys/syscall.h>
+pid_t gettid(void) { return syscall(SYS_gettid); }
+
+/* Assertion */
+#define refute_greater_or_equal(var, compare, reason)                          \
+    do                                                                         \
+    {                                                                          \
+        if ((var) >= (compare))                                                \
+        {                                                                      \
+            fprintf(stderr, ("[ERROR] event_profiling: " reason), var);        \
+            exit(1);                                                           \
+        }                                                                      \
+    } while (0)
```

Comment:
The usage of this macro seems the check against a runtime configuration, and doesn't feel appropriate to assert/abort.
I think it's better to raise an exception or just ignore.

Reply:
How about just printing to `stderr` but not `exit(1)`?

Reply:
I decide to remove all assertions